tools/autograph-edge/main.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/mozilla-services/yaml"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"go.mozilla.org/mozlogrus"
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/decrypt"
+)
+
+var (
+	ErrInvalidToken              = errors.New("invalid authorization token")
+	ErrInvalidMethod             = errors.New("only POST requests are supported")
+	ErrMissingBody               = errors.New("missing request body")
+	ErrAutographBadStatusCode    = errors.New("failed to retrieve signature from autograph")
+	ErrAutographBadResponseCount = errors.New("received an invalid number of responses from autograph")
+	ErrAutographEmptyResponse    = errors.New("autograph returned an invalid empty response")
+)
+
+func init() {
+	// initialize the logger
+	mozlogrus.Enable("autograph")
+}
+
+func main() {
+	if os.Getenv("LAMBDA_TASK_ROOT") != "" {
+		lambda.Start(Handler)
+	} else {
+		resp, err := Handler(events.APIGatewayProxyRequest{Body: os.Args[1]})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error: %v\n", err)
+			os.Exit(1)
+		}
+		log.Printf("%+v", resp)
+	}
+}
+
+func Handler(r events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
+	var headers string
+	for k, v := range r.Headers {
+		headers += k + ": " + v + "; "
+	}
+	log.WithFields(log.Fields{
+		"remoteAddressChain": "[" + r.Headers["X-Forwarded-For"] + "]",
```

Comment:
I assume lamdas sit behind an ELB thing that sets X-FF, but it'd be good to confirm this can't be spoofed.

Reply:
I assume it can be polluted and the elb will add values at the end, so this is similar to what we're doing in other places (don't trust the first value to be the public ip, count from the end).